core/arch/arm/kernel/user_ta.c

```diff
@@ -621,21 +504,410 @@ TEE_Result tee_ta_register_ta_store(struct user_ta_store_ops *ops)
 	return TEE_SUCCESS;
 }
 
-TEE_Result tee_ta_init_user_ta_session(const TEE_UUID *uuid,
-			struct tee_ta_session *s)
+static char tolowercase(char c)
+{
+	if (c >= 'A' && c <= 'Z')
+		return c - 'A' + 'a';
+	return c;
+}
+
+static int8_t hex(char c)
+{
+	char lc = tolowercase(c);
+
+	if (lc >= '0' && lc <= '9')
+		return (int8_t)(lc - '0');
+	if (lc >= 'a' && lc <= 'f')
```

Comment:
could use `if (isxdigit(lc))`, since the normal digits are taken care of already